stuf_zds_payments/plugin.py

```diff
@@ -0,0 +1,121 @@
+from stuf.stuf_zds.client import ZaakOptions
+
+from .client import get_client
+from .registration_variables import register as variables_registry
+
+logger = logging.getLogger(__name__)
+
+PLUGIN_IDENTIFIER = "stuf-zds-create-zaak:ext-utrecht"
+
+
+def default_payment_status_update_mapping() -> list[tuple[str, str]]:
+    return [
+        ("payment_completed", "payment_completed"),
+        (
+            "payment_amount",
+            "payment_amount",
+        ),
+        (
+            "payment_public_order_ids",
+            "payment_public_order_ids",
+        ),
+    ]
+
+
+def prepare_value(value: Any):
+    match value:
+        case bool():
+            return "true" if value else "false"
+        case float():
+            return str(value)
+        case _:
+            return value
+
+
+class ZaakPaymentOptionsSerializer(ZaakOptionsSerializer):
+    payment_status_update_mapping = serializers.ListField(
+        child=serializers.ListField(
+            child=serializers.CharField(), min_length=2, max_length=2
+        ),
+        label=_("payment status update variable mapping"),
+        help_text=_(
+            "This mapping is used to map the variable keys to keys used in the XML "
+            "that is sent to StUF-ZDS. Those keys and the values belonging to them in "
+            "the submission data are included in extraElementen."
+        ),
+        required=False,
+    )
+
+    @classmethod
+    def display_as_jsonschema(cls):
+        data = super().display_as_jsonschema()
+        # Workaround because drf_jsonschema_serializer does not pick up defaults
+        data["properties"]["payment_status_update_mapping"][
+            "default"
+        ] = default_payment_status_update_mapping()
+        return data
+
+
+@register(PLUGIN_IDENTIFIER)
+class StufZDSPaymentsRegistration(StufZDSRegistration):
+    verbose_name = _("StUF-ZDS (payments)")
+    configuration_options = ZaakPaymentOptionsSerializer
+
+    def get_extra_payment_variables(
+        self, submission: "Submission", options: ZaakOptions
+    ):
+        key_mapping = dict(options["payment_status_update_mapping"])
+        return {
+            key_mapping[variable.key]: prepare_value(variable.initial_value)
+            for variable in get_static_variables(
+                submission=submission,
+                variables_registry=variables_registry,
+            )
+            if variable.key
+            in ["payment_completed", "payment_amount", "payment_public_order_ids"]
+            and variable.key in key_mapping
+        }
+
+    def get_extra_data(
+        self, submission: Submission, options: ZaakOptions
+    ) -> dict[str, Any]:
+        """
+        Overridden to ensure the extra payment variables are sent as extraElementen
+        when creating the Zaak
+        """
+        data = super().get_extra_data(submission, options)
+        data.update(self.get_extra_payment_variables(submission, options))
+        return data
```

Comment:
```suggestion
        payment_extra = self.get_extra_payment_variables(submission, options)
        return {**data, **payment_extra}
```
a side-effect free alternative :)